crates/sui-cluster-test/src/cluster.rs

```diff
@@ -178,13 +179,31 @@ impl Cluster for LocalNewCluster {
                 .expect("Unable to parse indexer address")
         });
 
-        let mut cluster_builder = TestClusterBuilder::new()
-            .set_genesis_config(genesis_config)
-            .enable_fullnode_events();
+        let mut cluster_builder = TestClusterBuilder::new().enable_fullnode_events();
 
-        if let Some(epoch_duration_ms) = options.epoch_duration_ms {
-            cluster_builder = cluster_builder.with_epoch_duration_ms(epoch_duration_ms);
+        // Check if we already have a config directory that is passed
+        if let Some(config_dir) = options.config_dir.clone() {
+            // Load the config of the Sui authority.
+            let network_config_path = config_dir.join(SUI_NETWORK_CONFIG);
+
+            let network_config: NetworkConfig = PersistedConfig::read(&network_config_path)
+                .map_err(|err| {
+                    err.context(format!(
+                        "Cannot open Sui network config file at {:?}",
+                        network_config_path
+                    ))
+                })?;
+
+            cluster_builder = cluster_builder.set_network_config(network_config);
+            cluster_builder = cluster_builder.with_config_dir(config_dir);
+        } else {
+            cluster_builder = cluster_builder.set_genesis_config(genesis_config);
```

Comment:
genesis_config should be defined and created inside this branch

Reply:
The original reason genesis config wasn't defined in this branch was because there are multiple situations where sui-cluster-test can run. 

This branch is mainly the situation where a tempdirectory is generated for the cluster test and a genesis is spun up and store in the temp directory. 

This is why I had originally modified the `sui genesis` command and change the allocations there. I will instead create a different genesis that gets stored in the config.